lib/uksched/include/uk/sched.h

```diff
@@ -139,7 +139,7 @@ static inline void uk_sched_thread_woken(struct uk_thread *t)
 /**
  * Create a main thread from current context and call thread starter function
  */
-int uk_sched_start(struct uk_sched *sched);
+int uk_sched_start(struct uk_sched *sched) __nonnull;
```

Comment:
The function already uses `UK_ASSERT`, using `__nonnull` is redundant here and would generate avoidable warnings